rllib/algorithms/ppo/ppo_base_rl_module.py

```diff
@@ -24,8 +24,6 @@ def __init__(self, config: RLModuleConfig):
         self.config = config
         catalog = config.catalog
 
-        assert isinstance(catalog, PPOCatalog), "A PPOCatalog is required for PPO."
```

Comment:
Jun requested this change, because he'd like to write his own Catalog from time to time that does not inherit from PPOCatalog but only provides the build_.* fields

Reply:
We should have a check catalog function that checks for certain methods and attributes that will be used in ppo constuctor instead.